core/src/main/resources/error/error-classes.json

```diff
@@ -23,6 +23,12 @@
     ],
     "sqlState" : "22005"
   },
+  "CANNOT_INFER_DATE" : {
+    "message" : [
+      "Cannot infer date in schema inference when LegacyTimeParserPolicy is \"LEGACY\". Legacy Date formatter does not support strict date format matching which is required to avoid inferring timestamps and other non-date entries to date."
+    ],
+    "sqlState" : "22005"
```

Comment:
Is this the right sqlState code? Tests would not pass until I added a sqlState code

Reply:
Changed to 22007 for date format error